WikiWatershed/model-my-watershed
# Upgrade Node

## Overview

Upgrades Node to 12. This was originally done in anticipation of a new build process for the front-end, which has since been deemed extraneous. Yet, our node version is quite out of date and this upgrade has been a long time coming.

This aims to be a minimally disruptive update that gets us to the most current version without changing too much. It also fixes the Testem not exiting issue of #3110.

Connects #2323 
Connects #3115

### Demo

![image](https://user-images.githubusercontent.com/1430060/66786942-1118bf00-eeb0-11e9-8371-c5c6dfbc700f.png)

### Notes

The `ansible-yarn` module installs an older version of Node / NPM, which gets overwritten by `ansible-nodejs` in the next step, which is why Yarn is installed first. Ideally we'd be able to specify to the Yarn module which version of Node we want and just use that, but that is not currently possible. I have an open pull request for that https://github.com/azavea/ansible-yarn/pull/2.

## Testing Instructions

 * Check out this branch
 * Remove the `node_modules` folder `$ rm -rf src/mmw/node_modules`
 * Destroy the `app` VM and recreate it `$ vagrant destroy -f app && vagrant up`
 * Run the tests `$ ./scripts/test.sh`
    - [x] Ensure tests run
    - [x] Ensure script exits successfully
 * Go to [:8000/](http://localhost:8000/) and click around to ensure things still work correctly

## Checklist
